### PR TITLE
Add basic SquashFS support.

### PR DESCRIPTION
**Pull request by OscarL**
Only listing/extracting is implemented.

Tested on beta4, 32 and 64 bits. ~For proper usage, requires the following squashfs_tools PR to be merged: https://github.com/haikuports/haikuports/pull/10278 (mostly just to fix older version using the wrong endianness when creating/reading squashfs images).~ Needed fix already merged previously (https://github.com/haikuports/haikuports/pull/10274).

### CODE REVIEW

File: Source/CMakeLists.txt [hunk 1]
```diff
@@ -60,6 +60,7 @@ add_subdirectory(GZipArchiver)
 add_subdirectory(LhaArchiver)
 add_subdirectory(RarArchiver)
 add_subdirectory(TarArchiver)
+add_subdirectory(SquashFSArchiver)
```
Comment (augiedoggie) [inline Source/CMakeLists.txt:62]: Should be sorted alphabetically

Comment (OscarL) [reply Source/CMakeLists.txt:62]: Jeez, I DO need better/new glasses :-D

Done.

**Comment by augiedoggie** [comment]
I haven't tested it but it looks ok.  Can you create a quick page in the user guide at `Docs/UserGuide/addons/SquashFS.rst` and add it to `Docs/UserGuide/index.rst`?

File: Source/SquashFSArchiver/SquashFSArchiver.cpp [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+    m_pipeMgr << m_archivePath.Path();
+
+    for (int32 i = 0; i < count; i ++)
+    {
+        const char* pathString = NULL;
+        if (message->FindString(kPath, i, &pathString) == B_OK)
+            m_pipeMgr << SupressWildcards(pathString);
+    }
+
+    m_pipeMgr.PrintToStream();
```
Comment (OscarL) [inline Source/SquashFSArchiver/SquashFSArchiver.cpp]: Note to self: remove this.

**Pull request merged**